src/semaphore.c

```diff
@@ -716,6 +771,11 @@ _dispatch_thread_semaphore_create(void)
 	int ret = sem_init(s4, 0, 0);
 	DISPATCH_SEMAPHORE_VERIFY_RET(ret);
 	return (_dispatch_thread_semaphore_t)s4;
+#elif USE_FUTEX_SEM
+	dispatch_futex_t *futex = malloc(sizeof(*futex));
```

Comment:
actually I looked at the dispatch source and using a per thread static one doesn't work because the memory has to be reusable because of dispatch_apply and similar codepaths, so you need the malloc as sad as this is. though instead of doing malloc I would use a continuation as it will very often be taken from a very hot pool.

IOW just replace the malloc with a _dispatch_continuation_alloc(), use it (it's way larger than what you need) and free it with _dispatch_continuation_free().